File: 0454-4Sum II/0454-4Sum II.cpp
class Solution {
public:
    int fourSumCount(vector<int>& A, vector<int>& B, vector<int>& C, vector<int>& D) {
        unordered_map<int, int> table;
        for (int a : A) {
            for (int b : B) {
                ++table[a + b];
            }
        }
        
        int count = 0;
        for (int c : C) {
            for (int d : D) {
                count += table[-c-d];
            }
        }
        
        return count;
    }
};

File: 0401-0500/0454-4Sum II/0454-4Sum II.py
import collections
class Solution:
    def fourSumCount(self, A: List[int], B: List[int], C: List[int], D: List[int]) -> int:
        table = collections.Counter(a + b for a in A for b in B)
        return sum(table[-c-d] for c in C for d in D)
